File: test-wix-api.js
#!/usr/bin/env node

// Load environment variables from .env.local
require('dotenv').config({ path: '/Users/marcusbinnie/jt-physio/.env.local' });

const axios = require('axios');

console.log(`
╔════════════════════════════════════════════════════════════╗
║         WIX API - DIRECT AUTHENTICATION TEST              ║
╚════════════════════════════════════════════════════════════╝
`);

// Check credentials
console.log('📋 Checking credentials...\n');

const WIX_API_KEY = process.env.WIX_API_KEY;
const WIX_SITE_ID = process.env.WIX_SITE_ID;
const WIX_AUTHOR_MEMBER_ID = process.env.WIX_AUTHOR_MEMBER_ID;

if (!WIX_API_KEY) {
  console.log('❌ WIX_API_KEY is missing');
  process.exit(1);
}

if (!WIX_SITE_ID) {
  console.log('❌ WIX_SITE_ID is missing');
  process.exit(1);
}

if (!WIX_AUTHOR_MEMBER_ID) {
  console.log('❌ WIX_AUTHOR_MEMBER_ID is missing');
  process.exit(1);
}

console.log('✓ WIX_API_KEY: ' + WIX_API_KEY.substring(0, 20) + '...');
console.log('✓ WIX_SITE_ID: ' + WIX_SITE_ID);
console.log('✓ WIX_AUTHOR_MEMBER_ID: ' + WIX_AUTHOR_MEMBER_ID);

console.log(`
╔════════════════════════════════════════════════════════════╗
║      TEST 1: POST /blog/v3/posts/query (Query Posts)      ║
╚════════════════════════════════════════════════════════════╝
`);

axios({
  method: 'POST',
  url: 'https://www.wixapis.com/blog/v3/posts/query',
  headers: {
    // Wix API Key auth uses the raw key value (not Bearer)
    'Authorization': WIX_API_KEY,
    'wix-site-id': WIX_SITE_ID,
    'Content-Type': 'application/json'
  },
  data: {
    // Default sort is firstPublishedDate DESC with pinned first.
    // Keep it explicit and limit output.
    paging: { limit: 5, offset: 0 }
  }
})
  .then(response => {
    const posts = response.data.posts || [];

    console.log('✓ SUCCESS! Blog posts retrieved:');
    console.log(`  Posts count: ${posts.length}`);
    console.log(`  Status: ${response.status}\n`);

    if (posts.length > 0) {
      console.log('  Latest post:');
      const post = posts[0];
      console.log(`    Title: ${post.title || '(no title)'}`);
      console.log(`    ID: ${post.id}`);
      console.log(`    First published: ${post.firstPublishedDate || '(not published)'}`);
    }

    testCreatePost();
  })
  .catch(error => {
    console.log('❌ FAILED');
    console.log(`  Status: ${error.response?.status || 'Unknown'}`);
    console.log(`  Message: ${error.response?.statusText || error.message}`);
    console.log(`  Error: ${JSON.stringify(error.response?.data, null, 2)}`);
    console.log(`
DIAGNOSIS:
- If 404: Blog API endpoint not found (check URL) or Blog app not installed
- If 401: API key is invalid
- If 403: API key missing required blog permissions (e.g. Read Blog)

Next step: Check Wix Dashboard → Settings → API for permissions
`);
    process.exit(1);
  });

function testCreatePost() {
  console.log(`
╔════════════════════════════════════════════════════════════╗
║    TEST 2: POST /blog/v3/draft-posts (Create Draft)       ║
╚════════════════════════════════════════════════════════════╝
`);

  const testPost = {
    draftPost: {
      title: 'Test Blog Post - Knee Injury Recovery',
      slug: 'test-knee-injury-recovery-' + Date.now(),
      // For 3rd-party apps, memberId is required
      memberId: WIX_AUTHOR_MEMBER_ID,
      categoryIds: [],
      // Minimal Rich Content document
      richContent: {
        nodes: [
          {
            type: 'PARAGRAPH',
            id: 'p1',
            nodes: [
              {
                type: 'TEXT',
                id: 't1',
                text: 'This is a test blog post created by the JT Physiotherapy automation system. It demonstrates the integration with the Wix Blog API.'
              }
            ]
          }
        ],
        metadata: { version: 1 },
        documentStyle: {}
      },
      excerpt: 'Test blog post for Wix API validation'
    }
  };

  axios({
    method: 'POST',
    url: 'https://www.wixapis.com/blog/v3/draft-posts',
    headers: {
      // Wix API Key auth uses the raw key value (not Bearer)
      'Authorization': WIX_API_KEY,
      'wix-site-id': WIX_SITE_ID,
      'Content-Type': 'application/json'
    },
    data: testPost
  })
    .then(response => {
      const draft = response.data.draftPost;
      console.log('✓ SUCCESS! Test draft created:');
      console.log(`  Draft ID: ${draft.id}`);
      console.log(`  Title: ${draft.title}`);
      console.log(`  Slug: ${draft.slug}`);
      console.log(`  Status: ${response.status}\n`);
      
      console.log(`
✅ WIX API IS FULLY FUNCTIONAL!

Your blog automation system can now:
1. List existing blog posts
2. Create new blog posts
3. Update blog posts
4. Publish blog posts

✓ Credentials are valid
✓ Blog app is installed
✓ API permissions are correct
✓ Network connection is working

You're ready to use the blog automation system!
`);
    })
    .catch(error => {
      console.log('❌ FAILED');
      console.log(`  Status: ${error.response?.status || 'Unknown'}`);
      console.log(`  Message: ${error.response?.statusText || error.message}`);
      console.log(`  Error: ${JSON.stringify(error.response?.data, null, 2)}`);
      console.log(`
DIAGNOSIS:
- If 404: Blog app not installed or endpoint incorrect
- If 401: API key invalid
- If 403: Missing blog write permissions (e.g. Write Blog)
- If 400: Invalid request format (payload schema)

SOLUTIONS:
1. Install Blog app:
   Go to Wix App Market → Search "Blog" → Install

2. Check API permissions:
   Wix Dashboard → Settings → API → Ensure Read Blog / Write Blog are enabled

3. Verify credentials:
   - WIX_API_KEY in .env.local matches your Wix API key
   - WIX_SITE_ID matches your Wix site ID
   - WIX_AUTHOR_MEMBER_ID is a valid memberId

4. Test again:
   node test-wix-api.js
`);
      process.exit(1);
    });
}

File: test-wix-api.py
#!/usr/bin/env python3
import requests
import json

admin_password = "JtPhysio_Admin_2026!9kP"
base_url = "http://localhost:3000"

# Fetch list of posts
print("=" * 60)
print("Fetching list of published posts...")
print("=" * 60)
response = requests.get(f"{base_url}/api/blog/wix-posts", 
    headers={'Authorization': admin_password})
print(f"Status: {response.status_code}")
data = response.json()
print(f"Posts count: {len(data.get('posts', []))}")

if data.get('posts'):
    first_post = data['posts'][0]
    print(f"\nFirst post ID: {first_post.get('id')}")
    print(f"First post title: {first_post.get('title')}")
    print(f"First post fields: {list(first_post.keys())}")
    print(f"Has 'content' field: {'content' in first_post}")
    print(f"Has 'richContent' field: {'richContent' in first_post}")
    
    # Now try to fetch details for that post
    post_id = first_post.get('id')
    print(f"\n{'=' * 60}")
    print(f"Fetching details for post: {post_id}")
    print(f"{'=' * 60}")
    
    details_response = requests.get(
        f"{base_url}/api/blog/wix-posts?action=details&postId={post_id}",
        headers={'Authorization': admin_password}
    )
    print(f"Status: {details_response.status_code}")
    details_data = details_response.json()
    
    if 'error' in details_data:
        print(f"Error: {details_data['error']}")
    elif 'post' in details_data:
        post = details_data['post']
        print(f"Post title: {post.get('title')}")
        print(f"Post fields: {list(post.keys())}")
        print(f"Has 'richContent' field: {'richContent' in post}")
        print(f"Has 'htmlBody' field: {'htmlBody' in post}")
        print(f"Has 'plainContent' field: {'plainContent' in post}")
        
        if post.get('richContent'):
            print(f"\nrichContent type: {type(post['richContent'])}")
            if isinstance(post['richContent'], dict):
                print(f"richContent keys: {list(post['richContent'].keys())}")
                print(f"Has 'nodes' field: {'nodes' in post['richContent']}")
